### PR TITLE
Support multiple simulations and replicates in cloud handler

### PR DESCRIPTION
**Pull request by sampottinger**
## Summary
Enhanced the cloud simulation handler to support running multiple simulations and multiple replicates per simulation in a single request, while also adding URL-to-script preprocessing to optimize script transmission via query parameters.

## Key Changes

- **Multiple simulations support**: The `simulation` query parameter now accepts comma-separated scenario names (with automatic whitespace trimming), allowing users to run multiple scenarios in one request and receive combined CSV results.

- **Replicates parameter**: Added a new `replicates` query parameter that controls how many times each simulation runs. Defaults to 1 when omitted. Invalid values (non-integer, zero, or negative) return HTTP 400.

- **CloudQubectalkPreprocessor**: New utility class that replaces pipe characters (`|`) with spaces in incoming scripts. Since `|` doesn't appear in QubecTalk syntax, it serves as a compact URL-safe substitute for spaces, reducing query parameter size by ~66% for space-heavy scripts.

- **InvocationParameters refactoring**: 
  - Changed `simulation` from `Optional<String>` to `List<String>` to represent multiple scenarios
  - Added `replicates` field with validation-ready integer storage
  - Empty list indicates validate-only requests (no simulations to run)

- **InvocationParametersFactory updates**: 
  - Parses comma-separated simulation names with trimming
  - Parses and validates replicates as positive integer (stores 0 for invalid values)
  - Maintains backward compatibility with absent parameters

- **SimulationHandler logic**: 
  - Applies preprocessing to scripts before parsing
  - Validates replicates value and returns 400 for invalid inputs
  - Iterates through all requested simulations and replicates, collecting results into a single CSV response
  - Updated documentation to reflect multi-simulation capability

- **Comprehensive test coverage**: Added 10 new unit tests covering comma-separated simulations, whitespace handling, and all replicates validation scenarios. Added 6 new unit tests for the preprocessor covering edge cases.

- **CI/CD**: Added Lambda deployment job to GitHub Actions workflow for automated deployment to AWS when pushing to the `deploy` branch.

## Implementation Details

The handler now uses `flatMap` to generate results across all combinations of simulations and replicates, maintaining order. The preprocessor is instantiated once per handler instance and applied transparently before script parsing, ensuring backward compatibility with existing scripts while enabling more compact URL encoding for new requests.

https://claude.ai/code/session_018HJuAnQ89mZkJAjnZR3n7N

### CODE REVIEW

File: engine/src/main/java/org/kigalisim/cloud/InvocationParameters.java [hunk 1]
```diff
@@ -41,12 +52,28 @@ public Optional<String> getScript() {
...
 
   /**
-   * Returns the simulation name parameter.
+   * Returns the list of simulation names to run.
    *
-   * @return An {@link Optional} containing the simulation name, or empty if not provided.
+   * <p>Returns an empty list when no {@code simulation} query parameter was provided,
+   * indicating that the request is validate-only.</p>
+   *
+   * @return An unmodifiable {@link List} of simulation names; never {@code null}.
```
Comment (sampottinger) [inline engine/src/main/java/org/kigalisim/cloud/InvocationParameters.java]: No need to say never null. We will clarify that in CONTRIBUTING.md later.

File: engine/src/main/java/org/kigalisim/cloud/SimulationHandler.java [hunk 2]
```diff
@@ -101,28 +107,37 @@ private APIGatewayV2HTTPResponse handleRequestUnsafe(
...
+    for (String simulationParam : simulations) {
+      try {
+        resolvedNames.add(resolveScenarioName(program, simulationParam));
+      } catch (IllegalArgumentException e) {
+        return buildResponse(STATUS_BAD_REQUEST, CONTENT_TYPE_TEXT, e.getMessage());
+      }
+    }
+
+    List<EngineResult> results = resolvedNames.stream()
+        .flatMap(scenarioName ->
```
Comment (sampottinger) [inline engine/src/main/java/org/kigalisim/cloud/SimulationHandler.java]: This is kinda a lot. Let's use curly braces to make a body with a return on IntStream ending with `})`

File: engine/src/test/java/org/kigalisim/cloud/SimulationHandlerTest.java [hunk 3]
```diff
@@ -166,4 +166,154 @@ public void testUnknownSimParamReturns400() {
...
+   * Test that a comma-separated list of two valid simulation names returns HTTP 200.
+   */
+  @Test
+  public void testTwoSimsCommaSeparatedReturns200() {
+    Map<String, String> params = new HashMap<>();
+    params.put("script", twoScenarioScript);
+    params.put("simulation", "Business as Usual,Scenario Two");
+    APIGatewayV2HTTPEvent event = buildEvent(params);
+    APIGatewayV2HTTPResponse response = handler.handleRequest(event, null);
+    assertEquals(200, response.getStatusCode(),
```
Comment (sampottinger) [inline engine/src/test/java/org/kigalisim/cloud/SimulationHandlerTest.java]: Let's format these like:

```
assertEquals(
    200,
    response.getStatusCode(),
    "Comma-separated valid simulation names should return 200"
);
```
